modules/1_opencv/src/opencv_coins.cpp

```diff
@@ -1,14 +1,41 @@
 #include "opencv_coins.hpp"
+#include <vector>
+#define _USE_MATH_DEFINES
+#include <math.h>
 
 using namespace cv;
 
 unsigned countCoins(const Mat& img) {
-    Mat gray, thresh;
+	Mat gray, thresh, sure_bg, sure_fg, dist_transform, unknown, markedimg;
+	Mat kernel = Mat::ones(3, 3, CV_8U);
+	double max = -100000.0;
+	unsigned int result = 0;
     cvtColor(img, gray, COLOR_BGR2GRAY);
     threshold(gray, thresh, 0, 255, THRESH_BINARY_INV + THRESH_OTSU);
-    // Remove unfilled black holes by morphological closing
-    morphologyEx(thresh, thresh, MORPH_CLOSE, Mat::ones(3, 3, CV_8U), Point(-1, -1), 3);
-
-    // TODO: implement an algorithm from https://docs.opencv.org/master/d3/db4/tutorial_py_watershed.html
-    CV_Error(Error::StsNotImplemented, "countCoins");
+    morphologyEx(thresh, thresh, MORPH_CLOSE, kernel, Point(-1, -1), 3);			
+	dilate(thresh, sure_bg, kernel, Point(-1, -1), 3);
```

Comment:
Please use same count spaces per string

Reply:
I don't know what the problem is, but in VS it looks right 
![2020-07-08_13-55-26](https://user-images.githubusercontent.com/42999373/86910667-b3ee2900-c122-11ea-9b1a-35ca3cb6d0b7.png)

Reply:
try to use tabs as spaces in VS settings, and check with help `git diff`